docs/source/sandbox/packages.rst

```diff
@@ -11,40 +9,63 @@ providers etc), which we will call 'backends', and provide a very high level,
-terms will be Distribution(s) specific.
+We want to concentrate on (re)creation of such computation environments from a
+specification which is agnostic of a backend and concentrates on describing
+what constitutes the content of that environment relevant for the execution of
+computation.  Backend-specific details of construction, execution and
+interfacing with the backend should be "templated" (or otherwise parametrized
+in sufficient detail) so an advanced user could still provide their tune ups).
+We will not aim at the specification to be OS agnostic, i.e. the package
+configuration will have terms that are specific to an architecture or
+distribution.
 
 Construction of such environments would heavily depend on specification of
-"Distribution"s which contain sufficient information to reconstruct and
-execute in the environment. Such specifications could be constructed
-manually, by ReproNim from loose human description, or via automated
-provenance collection of "shell" command.  They also should provide
-sufficient expressive power to be able to tune them quickly for most common
-cases (e.g. upgrade from release X to release Y)
-
-Distributions
-=============
-
-We would like to be able to cover various "distributions" of software (and
-data). Distribution as such is just a collection of units, usually called
-packages, which allows to manage (install, uninstall, upgrade, ...) via a
-unified interface.
-
+"packages" which contain sufficient information to reconstruct and execute in
+the environment. Such specifications could be constructed manually, by ReproNim
+from loose human description, or via automated provenance collection of "shell"
+command.  They also should provide sufficient expressive power to be able to
+tune them quickly for most common cases (e.g. upgrade from release X to
+release Y)
+
+Packages, Package Managers, and Distributions
+=============================================
+
+We would like to be able to identify, record, and install various "packages" of
+software and data. A package is a collection of files, potentially platform
+specific (in the case of binary packages) or requiring reconstruction (such as
+compiling applications from source). In addition, installing a package may have
+dependencies (additional packages required by the initial package to correctly
+operate). 
+
+Packages are installed, removed, and queried through the use of "package
+managers." There are different package managers for different components of an
+environment and have slightly different capabilities.  For example, "yum" and
+"apt-get" are used to install binary and source files on a Linux operating
+system.  "pip" provides download and compilation capability for the Python
+interpreted language, while "conda" is another Python package manager that can
+supports "virtual environments" (essentially subdirectories) that provide
+separate parallel Python environments.
```

Comment:
see `Myth #2` of https://jakevdp.github.io/blog/2016/08/25/conda-myths-and-misconceptions/   ;)  so conda aims to provide more generic than just Python environments support